### PR TITLE
Fix flakey SQLite test

### PR DESCRIPTION
**Pull request by bricelam**

### CODE REVIEW

File: test/Microsoft.Data.Sqlite.Tests/SqliteTransactionTest.cs [hunk 1]
```diff
@@ -115,38 +115,80 @@ public void Serialized_disallows_dirty_reads()
...
+                {
+                    File.Delete("deferred.db");
                 }
+            }
+        }
 
-                using (var connection1 = new SqliteConnection(connectionString))
-                using (var connection2 = new SqliteConnection(connectionString))
+        [Fact]
+        public void Deferred_allows_parallel_reads_new()
```
Comment (smitpatel) [inline test/Microsoft.Data.Sqlite.Tests/SqliteTransactionTest.cs:157]: noncached/sharedcache rather than old/new?

Comment (bricelam) [reply test/Microsoft.Data.Sqlite.Tests/SqliteTransactionTest.cs:157]: Temporary. Just keeping the old one around to see when it fails but the new one doesn't.

Comment (bricelam) [reply test/Microsoft.Data.Sqlite.Tests/SqliteTransactionTest.cs:157]: I'm abusing the CI since I can't repro locally

**Comment by bricelam** [comment]
Alright, it passed over 100,000 times on the CI. I'm convinced it's fixed.

**Pull request merged**